### PR TITLE
Refactor and style suggestions.

### PR DESCRIPTION
**Pull request by rleonid**
I tried to simplify `get_path` and `load_exn`

[Review on Reviewable](https://reviewable.io/reviews/hammerlab/ketrew/151)

### CODE REVIEW

File: src/lib/ketrew_configuration.ml [hunk 1]
```diff
@@ -234,28 +234,21 @@ module File = struct
...
-               fun () -> if Sys.file_exists path then Some path else None))
-        (fun () -> ketrew_path ^ "configuration.json")
-    in
-    try_options [
-      env "KETREW_CONFIGURATION";
-      env "KETREW_CONFIG";
-    ]
-      find_in_path
+      ] default_ketrew_path in
+    try_options possible_files (ketrew_path ^ "configuration.json")
```
Comment (smondet) [inline src/lib/ketrew_configuration.ml]: I prefer my lazy-more-functional version; but both are bugged:

```ocaml
        (List.map default_configuration_filenames
           ~f:(fun path ->
               fun () -> if Sys.file_exists path then Some path else None))
```

→ forgot to add the computed path in front of the filename !

File: src/lib/ketrew_configuration.ml [hunk 2]
```diff
@@ -293,22 +286,23 @@ module File = struct
...
-    | Some "sh" ->
-      read_command_output_no_lwt_exn path
-      |> parse_string_exn
-    | Some "url" ->
-      failwith "Getting config from URL: not implemented"
-    | Some other ->
-      Log.(s "Config file should a discriminatory extension, not "
-           % quote other % s " pretending it is `.json`" @ warning);
-      read_file_no_lwt path |> parse_string_exn
+    let ext = try Filename.chop_extension path with Invalid_argument _ -> "" in
```
Comment (smondet) [inline src/lib/ketrew_configuration.ml]: Chop-extension does the contrary:

```ocaml
utop[0]> Filename.chop_extension "bouh.bah";;
- : bytes = "bouh"
```

Comment (rleonid) [inline src/lib/ketrew_configuration.ml]: Oh, right. I keep forgetting that. Add it to Nonstd?

File: src/lib/ketrew_configuration.ml (hunk 1 above)
Comment (rleonid) [inline src/lib/ketrew_configuration.ml]: Your version is unnecessarily lazy and confusing.

Comment (smondet) [inline src/lib/ketrew_configuration.ml]: But the order in which tests are performed is very important for the priority of env-variables.
(I added the label `~and_then` to the fixed version):

https://github.com/hammerlab/ketrew/tree/fix-get-configuration-path

Comment (rleonid) [inline src/lib/ketrew_configuration.ml]: Unless `List.map` doesn't perform it's checks in order. Listing the `env` calls first establishes the appropriate order?

**Pull request closed without merging**